asv_bench/benchmarks/gil.py

```diff
@@ -320,3 +325,21 @@ def time_nogil_kth_smallest(self):
         def run(arr):
             algos.kth_smallest(arr, self.k)
         run()
+
+
+class nogil_read_csv(object):
+    number = 1
+    repeat = 5
+
+    def setup(self):
+        if (not have_real_test_parallel):
+            raise NotImplementedError
+        # Using the values
+        self.data = '0.1213700904466425978256438611,0.0525708283766902484401839501,0.4174092731488769913994474336\n        0.4096341697147408700274695547,0.1587830198973579909349496119,0.1292545832485494372576795285\n        0.8323255650024565799327547210,0.9694902427379478160318626578,0.6295047811546814475747169126\n        0.4679375305798131323697930383,0.2963942381834381301075609371,0.5268936082160610157032465394\n        0.6685382761849776311890991564,0.6721207066140679753374342908,0.6519975277021627935170045020\n        '
```

Comment:
I would just do use random here. e.g. ``DataFrame(np.random.randn(10000,10))`` (or whatever makes this a reasonable time

also can you add a tests that uses object and another that has a datetime parse in the index_col